lib/IDE/TypeContextInfo.cpp

```diff
@@ -104,12 +104,17 @@ void ContextInfoCallbacks::doneParsing() {
   for (auto T : Info.getPossibleTypes()) {
     if (T->is<ErrorType>() || T->is<UnresolvedType>())
       continue;
-    if (auto env = CurDeclContext->getGenericEnvironmentOfContext())
-      T = env->mapTypeIntoContext(T);
+
+    T = T->getRValueType();
+    if (T->hasArchetype())
```

Comment:
Is this ever true? In the old code we would map into context unconditionally. Or is env always null?

Reply:
Yes, it just used to lack tests for generic environment 😞

Reply:
Gotcha.